dsl/com/electriccloud/commander/dsl/util/BaseObject.groovy

```diff
@@ -225,6 +225,8 @@ abstract class BaseObject extends DslDelegatingScript {
     GroovyShell sh = new GroovyShell(this.scriptClassLoader, bindingMap? new Binding(bindingMap) : new Binding(), cc);
     DelegatingScript script = (DelegatingScript)sh.parse(new File(dslFile))
     script.setDelegate(this.delegate);
+    // add bindingMap to DslDelegate to deal with collections removing in 'overwrite' mode
+    script.getDelegate().getBinding().getVariables().put("bindingMap", bindingMap)
```

Comment:
following groovy conventions, this can be replaced with `script.delegate.binding.setVariable("bindingMap", bindingMap)`

Reply:
done